daft/table/table.py

```diff
@@ -114,14 +114,39 @@ def concat(cls, to_merge: list[Table]) -> Table:
     ###
 
     def to_arrow(self) -> pa.Table:
-        python_fields = [field for field in self.schema() if field.dtype == DataType.python()]
+        python_fields = {field.name for field in self.schema() if field.dtype == DataType.python()}
         if python_fields:
-            raise ValueError(f"Cannot convert table to Arrow due to presence of Python columns f{python_fields}")
-        return pa.Table.from_batches([self._table.to_arrow_record_batch()])
+            table = {}
+            for colname in self.column_names():
+                column_series = self.get_column(colname)
+                if colname in python_fields:
+                    # TODO(Clark): Get the column as a top-level ndarray to ensure it remains contiguous.
+                    column = column_series.to_pylist()
+                    # TODO(Clark): Infer the tensor extension type even when the column is empty.
+                    # This will probably require encoding more information in the Daft type that we use to
+                    # represent tensors.
+                    if len(column) > 0 and isinstance(column[0], np.ndarray):
```

Comment:
Do we implicitly pass-through the column on the `else` case here then? Should we instead throw a useful error like "Python column cannot be parsed to arrow"?

Reply:
This goes through the best-effort Arrow conversion path, as I commented here: https://github.com/Eventual-Inc/Daft/pull/837/files#r1174146060

If the column is empty, it will create a null-typed column, which is idiomatic for Arrow.

If the column is non-empty but holds non-tensor data, Arrow will do it's typical type inference and try to represent the data with one of its types, throwing an error if that's not possible.